src/Umbraco.Infrastructure/Examine/IndexRebuildStatusManager.cs

```diff
@@ -0,0 +1,28 @@
+using Examine;
+
+namespace Umbraco.Cms.Infrastructure.Examine;
+
+/// <summary>
+///
+/// </summary>
+public class IndexRebuildStatusManager : IIndexRebuildStatusManager
```

Comment:
My recommended approach would be to do this: https://github.com/Shazwazza/Examine/issues/372#issuecomment-1889733492

The only real way to know if the indexing is done in a resilient way would be to have an actual document in the index certifying that rebuilding is successful instead of relying on in-memory cache which is problematic.

Reply:
@bielu Sorry, I probably wasn't clear in my suggestion. We don't want to use an extra index to store any data, we can just use a marker document within the index. For example:

* Rebuilding an index deletes all data
* The index is populated with the normal data
* When the IndexPopulator is done populating the index, it then writes a special marker document signaling that the populator is done. Perhaps this document has a field like `__Populated: y`

Then the rebuild checker, just checks if the document count for `__Populated: y` == 1

Reply:
@Shazwazza that's make sense now! We can also extend it to check what populator are registered to show how many of them is done! I will make update to this pr

Reply:
@Shazwazza i started changing implementation of this service to use examine underhood, can you have quick look and check if that is what you had in mind? 
as in this way now we can also repeat failed batches (but i think I will need play around little more)